frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -59,6 +59,8 @@ local ReaderDictionary = InputContainer:new{
     data_dir = nil,
     dict_window_list = {},
     disable_lookup_history = G_reader_settings:isTrue("disable_lookup_history"),
+    dicts_order = G_reader_settings:readSetting("dicts_order", {}),
+    dicts_disabled = G_reader_settings:readSetting("dicts_disabled", {}),
```

Comment:
Yeah, those references are unique, as they should, but that one is a value, I'll move it to init, good catch ;).

Not quite used to OOP, but the idea is the the "class" object is only initialized on require, right?

Reply:
Actually, no, they're not: the ref is broken and updated in `showDictionariesMenu` during the SortWidget shenanigans.

Reply:
> Not quite used to OOP, but the idea is the the "class" object is only initialized on require, right?

Yes, the one we create in each module with:
`local ReaderDictionary = InputContainer:new{???}`
and that we return, and what we return (any object) is cached by Lua module/require subsystem - and just given from the cache for subsequent require() of the same module.

But Lua OOP (or how the kind of way of doing it we use in KOReader) is a bit special:
We use the require-cached-object's `:new()` method - that is found far away in frontend/ui/widget/widget.lua `Widget:new(o)`: this method accepts an empty table, setmetatable it to the original require-cached-object (so stuff not found in its table are looked also in the original require-cached-object - and then call :init() which is defined in the near require-cached-object - and this :init might stuff to the normal table...

The side effect with the setmetatable stuff picking missing stuff (nil) from the near object in the remote require-cached-object  is that, if in some bit of code, we do `self.some_setting = nil`, we'll erase it only from the near object: it will stay in the remote require-cached-object   and will be found there if defined.

Some kind of object properties shadowing/unshadowing upper super class properties...